selfdrive/boardd/panda.cc

```diff
@@ -363,57 +363,47 @@ uint8_t Panda::len_to_dlc(uint8_t len) {
   }
 }
 
-void Panda::pack_can_buffer(const capnp::List<cereal::CanData>::Reader &can_data_list,
-                         std::function<void(uint8_t *, size_t)> write_func) {
-  if (send.size() < (can_data_list.size() * CANPACKET_MAX_SIZE)) {
-    send.resize(can_data_list.size() * CANPACKET_MAX_SIZE);
+static void write_packet(uint8_t *dest, int *write_pos, const uint8_t *src, size_t size) {
+  for (int i = 0, &pos = *write_pos; i < size; ++i, ++pos) {
+    if (pos % USBPACKET_MAX_SIZE == 0) {
+      dest[pos++] = pos / USBPACKET_MAX_SIZE;
```

Comment:
```suggestion
      dest[pos] = pos / USBPACKET_MAX_SIZE;
      pos++;
```
This is MISRA violation. We are not using MISRA for OP, but I want code related to panda be as close as possible to MISRA rules.